### ISSUE TITLE
 Dataset Selector: Error on Layer Change when Project is created with Authentification Configuration

### ISSUE THREAD

**Issue by signedav** [opengisch/QgisModelBaker#528]
When the source is like this, everything works fine:
```
QgsDataSourceUri(iface.activeLayer().dataProvider().dataSourceUri())
<QgsDataSourceUri: dbname='bakery' host=localhost user='postgres' password='postgres' key='t_id' checkPrimaryKeyUnicity='1' table="baskets_nobaskets10"."haltestelle">
```
When the source is like that, it trows an error:
```
QgsDataSourceUri(iface.activeLayer().dataProvider().dataSourceUri())
<QgsDataSourceUri: dbname='bakery' host=localhost authcfg=x697iks key='t_id' checkPrimaryKeyUnicity='1' table="baskets_nobaskets07"."haltestelle">
```

Error:
**I cannot reproduce anymore, since when I create with the authentification config, I receive another error:**
```
    return PGConnector(uri, schema)
  File "/home/dave/.local/share/QGIS/QGIS3/profiles/MacMoneysack/python/plugins/QgisModelBaker/libqgsprojectgen/dbconnector/pg_connector.py", line 43, in __init__
    self.conn = psycopg2.connect(uri)
  File "/home/dave/.local/lib/python3.8/site-packages/psycopg2/__init__.py", line 126, in connect
    dsn = _ext.make_dsn(dsn, **kwargs)
  File "/home/dave/.local/lib/python3.8/site-packages/psycopg2/extensions.py", line 153, in make_dsn
    parse_dsn(dsn)
psycopg2.ProgrammingError: invalid dsn: invalid connection option "authcfg"

```

**Comment by signedav** [comment]
And when having a new layer loaded from PG (not with MB but manually), this error occurs. It's all about the same I guess

(Error found by Romedi):
```
QgisModelBaker.libqgsprojectgen.dbconnector.db_connector.DBConnectorError: fe_sendauth: no password supplied

Traceback (most recent call last):
  File "C:/Users/rfill/AppData/Roaming/QGIS/QGIS3\profiles\Standard_3_16/python/plugins\QgisModelBaker\libqgsprojectgen\dbconnector\pg_connector.py", line 43, in __init__
    self.conn = psycopg2.connect(uri)
  File "C:\PROGRA~1\QGIS3~1.16\apps\Python37\lib\site-packages\psycopg2\__init__.py", line 126, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
psycopg2.OperationalError: fe_sendauth: no password supplied


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:/Users/rfill/AppData/Roaming/QGIS/QGIS3\profiles\Standard_3_16/python/plugins\QgisModelBaker\gui\panel\dataset_selector.py", line 133, in set_current_layer
    config_manager.get_uri(), configuration.dbschema
  File "C:/Users/rfill/AppData/Roaming/QGIS/QGIS3\profiles\Standard_3_16/python/plugins\QgisModelBaker\libqgsprojectgen\db_factory\pg_factory.py", line 34, in get_db_connector
    return PGConnector(uri, schema)
  File "C:/Users/rfill/AppData/Roaming/QGIS/QGIS3\profiles\Standard_3_16/python/plugins\QgisModelBaker\libqgsprojectgen\dbconnector\pg_connector.py", line 45, in __init__
    raise DBConnectorError(str(e), e)
QgisModelBaker.libqgsprojectgen.dbconnector.db_connector.DBConnectorError: fe_sendauth: no password supplied


Python-Version: 3.7.0 (v3.7.0:1bf9cc5093, Jun 27 2018, 04:59:51) [MSC v.1914 64 bit (AMD64)]
QGIS-Version: 3.16.4-Hannover Hannover, 654e76b3cb

Python-Pfad:

·                     C:/PROGRA~1/QGIS3~1.16/apps/qgis-ltr/./python

·                     C:/Users/rfill/AppData/Roaming/QGIS/QGIS3\profiles\Standard_3_16/python

·                     C:/Users/rfill/AppData/Roaming/QGIS/QGIS3\profiles\Standard_3_16/python/plugins

·                     C:/PROGRA~1/QGIS3~1.16/apps/qgis-ltr/./python/plugins

·                     C:\PROGRA~1\QGIS3~1.16\apps\Python37

·                     C:\PROGRA~1\QGIS3~1.16\apps\Python37\Scripts

·                     C:\PROGRA~1\QGIS3~1.16\bin\python37.zip

·                     C:\PROGRA~1\QGIS3~1.16\apps\Python37\DLLs

·                     C:\PROGRA~1\QGIS3~1.16\apps\Python37\lib

·                     C:\PROGRA~1\QGIS3~1.16\bin

·                     C:\PROGRA~1\QGIS3~1.16\apps\Python37\lib\site-packages

·                     C:\PROGRA~1\QGIS3~1.16\apps\Python37\lib\site-packages\win32

·                     C:\PROGRA~1\QGIS3~1.16\apps\Python37\lib\site-packages\win32\lib

·                     C:\PROGRA~1\QGIS3~1.16\apps\Python37\lib\site-packages\Pythonwin

·                     C:/Users/rfill/AppData/Roaming/QGIS/QGIS3\profiles\Standard_3_16/python

·                     C:\Users\rfill\AppData\Roaming\QGIS\QGIS3\profiles\Standard_3_16\python\plugins\DigitizingTools\tools

·                     C:\Users\rfill\AppData\Roaming\QGIS\QGIS3\profiles\Standard_3_16\python\plugins\DigitizingTools
```

**Comment by signedav** [comment]
Close this if #539 is merged - the rest is reported already here #472
